torch/distributed/distributed_c10d.py

```diff
@@ -2365,7 +2365,7 @@ def batch_isend_irecv(p2p_op_list):
     _check_p2p_op_list(p2p_op_list)
     group = p2p_op_list[0].group
     device = p2p_op_list[0].tensor.device
-    if device.type == "cuda":
+    if device.type == "cuda" or device.type == torch._C._get_privateuse1_backend_name():
```

Comment:
nit: not all backends using privateuse1 key will be gauranteed to support coalescing.  so this is more of a hack IMO.  I am not sure if there is a very clean alternative that is easy to write.  cc @fduwjj 

a cleaner approach would be to let backends define a property 'supports batch', or, let backends call a new torch API that lets torch know the backend supports batching.  i'm not sure how generalized it needs to be though...

@albanD wdyt

Reply:
> nit: not all backends using privateuse1 key will be gauranteed to support coalescing. so this is more of a hack IMO. I am not sure if there is a very clean alternative that is easy to write. cc @fduwjj
> 
> a cleaner approach would be to let backends define a property 'supports batch', or, let backends call a new torch API that lets torch know the backend supports batching. i'm not sure how generalized it needs to be though...
> 
> @albanD wdyt

I added a method in the base class to determine if the backend supports coalescing ?